pkg/vpc/cleanup.go

```diff
@@ -75,6 +75,8 @@ func CleanupNetworkInterfaces(ec2API ec2iface.EC2API, spec *api.ClusterConfig) e
 			return errors.Wrapf(err, "unable to delete network interface %q", eniID)
 		}
 		logger.Debug("deleted network interface %q", eniID)
+		// why we are removing only the first eni id
```

Comment:
@marccarre need your help to clarify this one :)

Reply:
or @martina-if if you can clarify this concern :)

Reply:
It looks like a bug to me. Actually I have seen deletion issues sometimes... it could be because of this...

Reply:
@cPu1 do you know anything about this?

Reply:
seems like this one requires some more analysis, let me raise one issue for tracking purpose. We can fix it in subsequent PR if you are fine :)

Reply:
I think that's a good idea, since it will be changing behavior :+1: then we can track it in a separate PR and if something breaks we know where to look

Reply:
cool, let me know if you have other comments, happy to address :)